### PR TITLE
logger: use eprintln! if the logger is not initialized

### PR DESCRIPTION
**Pull request by ethercflow**
ref #11651

Signed-off-by: Wenbo Zhang <ethercflow@gmail.com>

### What is changed and how it works?

Close #11651 

What's Changed:

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)
- No code

Side effects

- None

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by NingLin-P — APPROVED** [review]

**Review by 5kbpers — APPROVED** [review]

**Comment by BusyJay** [comment]
Is it possible to register a term logger first, and then change it to a configured one? So we can always use `info/debug/error`.

File: src/config.rs [hunk 1]
```diff
@@ -2827,17 +2827,17 @@ impl TiKvConfig {
...
         let default_log_cfg = LogConfig::default();
         if self.log_level != default_tikv_cfg.log_level {
-            println!("deprecated configuration, log-level has been moved to log.level");
+            eprintln!("deprecated configuration, log-level has been moved to log.level");
             if self.log.level == default_log_cfg.level {
-                println!("override log.level with log-level, {:?}", self.log_level);
+                eprintln!("override log.level with log-level, {:?}", self.log_level);
                 self.log.level = self.log_level;
             }
             self.log_level = default_tikv_cfg.log_level;
```
Comment (BusyJay) [inline src/config.rs:2835]: I don't get it. If `log_level` is set, then user must have customized the configuration, so it should be respected. The old configuration is more likely to be a stale one. Why overwrite a customized one with a stale one?

Comment (ethercflow) [reply src/config.rs:2835]: > I don't get it. If `log_level` is set, then user must have customized the configuration, so it should be respected. The old configuration is more likely to be a stale one. Why overwrite a customized one with a stale one?

I agree with you, but I think the current code matches your idea.  I don't understand which is the stale one?

Comment (BusyJay) [reply src/config.rs:2835]: I mis-read the line. What's the point of overwriting it with the default value?

**Review by BusyJay — APPROVED** [review]

**Pull request merged**